driver/src/main/java/org/neo4j/driver/internal/cluster/RediscoveryImpl.java

```diff
@@ -291,8 +293,9 @@ private ClusterComposition handleRoutingProcedureError( Throwable error, Routing
         // Retriable error happened during discovery.
         DiscoveryException discoveryError = new DiscoveryException( format( RECOVERABLE_ROUTING_ERROR, routerAddress ), error );
         Futures.combineErrors( baseError, discoveryError ); // we record each failure here
-        logger.warn( format( "Received a recoverable discovery error with server '%s', will continue discovery with other routing servers if available.",
-                routerAddress ), discoveryError );
+        String warningMessage = format( RECOVERABLE_DISCOVERY_ERROR_WITH_SERVER, routerAddress );
+        logger.warn( warningMessage );
+        logger.debug( warningMessage, discoveryError );
```

Comment:
It always feels kinda odd tbh to have a conditional debug log hidden away. Normally I would wrap this is in `if(debugIsEnabled)`, but our logger does that already internally…

Reply:
I actually expect loggers to skip this if debug is not enabled, but this is probably because I tended to work with SLF4J compliant loggers. In any case, our logger does that as you mentioned.